### PR TITLE
SL-18330: Use new standalone llsd module instead of llbase.llsd.

### PR DESCRIPTION
**Pull request by nat-goodspeed**
Ongoing development has moved to [standalone llsd](https://github.com/secondlife/python-llsd) instead of llbase.llsd. In particular, [the new llsd.parse()](https://github.com/secondlife/python-llsd/pull/4) has greatly improved ability to correctly parse the stdin stream from the viewer.

### CODE REVIEW

**Comment by nat-goodspeed** [comment]
n.b. 

    RuntimeError: The Poetry configuration is invalid:
            - [extras.pipfile_deprecated_finder.2] 'pip-shims<=0.3.4' does not match '^[a-zA-Z-_.0-9]+$'

Not an actual problem with this PR.

**Comment by nat-goodspeed** [comment]
Merged [the llsd PR](https://github.com/secondlife/python-llsd/pull/4), essential for this fix, but the pre-commit check is still busted due to invalid Poetry configuration -- not the code.

**Review by simon-linden — APPROVED** [review]
Some of the most important one line commits I've seen in a while - nice to have that self-contained!

**Comment by bennettgoble** [comment]
~@nat-goodspeed I suspect the pre-commit action is failing due to a cached dependency conflicting with a newer version of python that is getting pulled in by pre-commit.yaml's setup-python step. You could pin the python version used to[ the last CI run of pre-commit that passed](https://github.com/secondlife/leap/actions/runs/4189116695/jobs/7261058962), 3.11.1 to possibly solve the CI issue without disabling linting/tests.~ (See latest comment)

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,13 +0,0 @@
...
-  hooks:
-    - id: check-ast
-    - id: check-yaml
-    - id: mixed-line-ending
-    - id: trailing-whitespace
-- repo: https://github.com/pycqa/isort
-  rev: 5.10.1
-  hooks:
-    - id: isort
-      args: [--line-length=120]
```
Comment (bennettgoble) [inline .pre-commit-config.yaml:13]: Even if the CI/CD configuration is being removed I don't see why the pre-commit configuration itself needs to be removed.

File: .github/workflows/pre-commit.yaml [hunk 2]
```diff
@@ -1,18 +0,0 @@
...
-
-
-jobs:
-  pre-commit:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-python@v4
-        with:
-          python-version: 3.x
```
Comment (bennettgoble) [inline .github/workflows/pre-commit.yaml:17]: Suggestion: pin to 3.11.1

**Comment by bennettgoble** [comment]
My previous suggestion is not correct, it looks like one of the linters specified in the pre-commit-config had an incompatibility with the latest version of poetry. Fixed in #15. Once this is merged, rebase and things should be good.

**Review by bennettgoble — CHANGES_REQUESTED** [review]
Requires #15

**Comment by nat-goodspeed** [comment]
Closing: I'll resubmit without the flailing commits attempting to disable the pre-commit check.

**Pull request closed without merging**